tests/qnn/test_keras.py

```diff
@@ -534,6 +554,23 @@ def test_train_model(self, model, batch_size, n_qubits, output_dim):
 
         model.fit(x, y, batch_size=batch_size, verbose=0)
 
+    @pytest.mark.parametrize("n_qubits, output_dim", indices_up_to(3, has_tuple=True))
+    @pytest.mark.parametrize("batch_size", [2])
+    def test_train_model_for_density_matrix(self, model, batch_size, n_qubits, output_dim):
+        """Test if a model can train using the KerasLayer when QNode returns a density_matrix().
+        The model is composed of two KerasLayers sandwiched between Dense neural network layers,
+        and the dataset is simply input and output vectors of zeros."""
+
+        if not qml.tape_mode_active():
+            pytest.skip("This functionality is only supported in tape mode.")
```

Comment:
```suggestion
```
Tape mode is now core :smile:

Reply:
I added this because there are cases in the tests where tape mode is disabled which gives the following error when this particular test is run: 

`AttributeError: module 'pennylane' has no attribute 'density_matrix'`

Should I always enable tape mode when creating the qnode in the new `get_circuit_dm` fixture in order to fix this?

Reply:
Ah, my mistake then, happy to keep as is.